openage/convert/singlefile.py

```diff
@@ -297,45 +297,29 @@ def terrain_convert(drs, slp_path, palette_index, output_path, interfac=None):
 
     # import here to prevent that the __main__ depends on SLP
     # just by importing this singlefile.py.
-    from .slp import determine_rgba_matrix, SLP, SLPFrame, FrameInfo
+    from .slp import SLP, SLPFrame, FrameInfo
     from PIL import Image
-    from .texture import Texture, TextureImage
+    from .texture import Texture, TextureImage, merge_frames
     from .terrain_convert import Terrain_convert
     import numpy as np
     
 
     # parse the slp image
     info("parsing slp image...")
     slp_image = SLP(slp_file.read())
+    texture_tomod = Texture(slp_image, palette)
     
-    # In order to extract the special image data from the SLP, I have to merge first all the frames ?
-    # In this case, the format is bad, we have a ndarray(10,10,dtype=np.ndarray(49,97),ndtype=tuple4), where each ndarray is of
-    # size 49x97x4
-    # How do we need to pass "special SLP image data" to PIL Image ?
-    # Should we have a matrix of 490x970 with each pixel is a rgba tuple?
-    info("parsing SLP Image into RGBA Matrix")
-    image = np.empty([10,10],dtype=np.ndarray)
-    row=0
-    column=0
-    count=0
-    while count < 100:
-        if row % 10 ==0 and row > 0:
-            column+=1
-            row=0
-        image[row][column] = slp_image.main_frames[count].get_picture_data(palette)
-        row+=1
-        count+=1
 
     #matrix = determine_rgba_matrix(image, None, 0) ???
-    image_new = Image.fromarray(image)
+    image_new = texture_tomod.image_data.get_data().transpose(1,0,2)
```

Comment:
why transpose the array here?

Reply:
Everytime I use the function Image.fromarray(image_array, "RGBA"), the array is transposed (x and y are swapped). If I remember well my computer architecture courses, this is because it depends on the array representation in memory. ([https://github.com/python-pillow/Pillow/issues/2619#issuecomment-315333880](url)) I don't know if I'm right, but we should be able to identify which array representation is used in memory.

Reply:
Are unit SLPs transposed too? This bug was fixed some time ago and we've been using `Image.fromarray()` to save the textures without problems for a while.